rest/rest.go

```diff
@@ -45,16 +48,91 @@
 	"google.golang.org/grpc/credentials/insecure"
 )
 
-var log *logrus.Entry
+var (
+	log *logrus.Entry
+
+	// srv holds the global http.Server of our REST API.
+	srv *http.Server
+
+	// sock holds the listener socket of our REST API.
+	sock net.Listener
+
+	ready = make(chan bool)
+
+	// cors holds the global CORS configuration.
+	cors *corsConfig
+)
+
+// corsConfig holds all necessary configuration options for Cross-Origin Resource Sharing of our REST API.
+type corsConfig struct {
+	// allowedOrigins contains a list of the allowed origins
+	allowedOrigins []string
+
+	// allowedHeaders contains a list of the allowed headers
+	allowedHeaders []string
+
+	// allowedMethods contains a list of the allowed methods
+	allowedMethods []string
+}
+
+// CORSConfigOption represents functional-style options to modify the CORS configuration in RunServer.
```

Comment:
Is there a convention for naming function types with suffix `Func` or similar?

Reply:
Probably, but in this case we want to have a functional-style option and the convention there is to append `Option` to it and "hide" a little bit that this is a function type for user convenience.